test/e2e/knative_serving_test.go

```diff
@@ -301,3 +303,45 @@ func waitForRouteServingText(t *testing.T, caCtx *test.Context, routeDomain, exp
 		t.Fatalf("The Route at domain %s didn't serve the expected text \"%s\": %v", routeDomain, expectedText, err)
 	}
 }
+
+func testKnativeServingForGlobalProxy(t *testing.T, caCtx *test.Context) {
+	if err := test.UpdateGlobalProxy(caCtx, "http://1.2.4.5:8999"); err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+
+	t.Run("wait for it to be ready after update", func(t *testing.T) {
+		test.GetPod(caCtx, knativeServing)
+	})
+
+	t.Run("deploy knative service after proxy update", func(t *testing.T) {
+		if _, err := test.WithServiceReady(caCtx, proxyHelloworldService, testNamespace, proxyImage); err == nil {
+			t.Fatal("Knative Service not ready", err)
+		}
+	})
+
+	cfg, err := test.GetConfiguration(caCtx, proxyHelloworldService, testNamespace)
+	for _, cond := range cfg.Status.Conditions {
+		// After global proxy update every call goes through proxy server
+		// Here it give unable to pull image because it tries to connect to not running http server
+		if !strings.Contains(cond.Message, "dial tcp 1.2.4.5:8999: i/o timeout") {
```

Comment:
I'm curious if this message is different if the traffic doesn't go through the proxy. In other words, are we verifying here that we use the proxy?

Reply:
Yes this is to verify call goes via proxy or not 

and if proxy server not able to connect this is the error we get 
```
'Revision "helloworld-go-2fbt6" failed with message: Unable to fetch
      image "gcr.io/knative-samples/helloworld-go": failed to resolve image to digest:
      failed to fetch image information: Get https://gcr.io/v2/: proxyconnect tcp:
      dial tcp 1.2.4.5:8999: i/o timeout.'
```